src/check_subid_range.c

```diff
@@ -36,11 +36,9 @@ int main(int argc, char **argv)
 	owner = argv[1];
 	check_uids = argv[2][0] == 'u';
 	errno = 0;
-	start = strtoul_noneg(argv[3], NULL, 10);
-	if (errno != 0)
+	if (str2ul(&start, argv[3]) == -1)
```

Comment:
When calling `strtoimax()` or `strtoumax()` what's the meaning of base 0? That's what you are implicitly calling at this point

Reply:
```
$ MANWIDTH=72 man strtoimax | sed -n '/\(DESCRIPTION\|LIBRARY\|STANDARDS\|HISTORY\)/,/^$/p'
LIBRARY
     Standard C library (libc, -lc)

DESCRIPTION
     These  functions  are  just  like strtol(3) and strtoul(3), except
     that they return a value of type intmax_t and  uintmax_t,  respec‐
     tively.

STANDARDS
     C11, POSIX.1‐2008.

HISTORY
     POSIX.1‐2001, C99.

```

And here's some extract from strtol(3):

```
$ MANWIDTH=72 man strtol | sed -n '/DESCRIPTION/,+10p'
DESCRIPTION
     The strtol() function converts the initial part of the  string  in
     nptr  to  a  long integer value according to the given base, which
     must be between 2 and 36 inclusive, or be the special value 0.

     The string may begin with an arbitrary amount of white  space  (as
     determined by isspace(3)) followed by a single optional '+' or '-'
     sign.   If  base is zero or 16, the string may then include a "0x"
     or "0X" prefix, and the number will be read in base 16; otherwise,
     a zero base is taken as 10 (decimal) unless the next character  is
     '0', in which case it is taken as 8 (octal).
```

Let's see strtol(3):

Reply:
Which TL;DR means:

A base of 0 means that:

- `"0x1234"` will be interpreted as base 16
- `"01234"` will be interpreted as base 8
- `"1234"` will be interpreted as base 10
(And since C23, `"0b1234" will be interpreted as base 2.)

Reply:
If you prefer to keep it with a base 10, we can call a2ul() instead.

Reply:
> $ MANWIDTH=72 man strtol | sed -n '/DESCRIPTION/,+10p'
> DESCRIPTION
>      The strtol() function converts the initial part of the  string  in
>      nptr  to  a  long integer value according to the given base, which
>      must be between 2 and 36 inclusive, or be the special value 0.
> 
>      The string may begin with an arbitrary amount of white  space  (as
>      determined by isspace(3)) followed by a single optional '+' or '-'
>      sign.   If  base is zero or 16, the string may then include a "0x"
>      or "0X" prefix, and the number will be read in base 16; otherwise,
>      a zero base is taken as 10 (decimal) unless the next character  is
>      '0', in which case it is taken as 8 (octal).
> ```

I was unable to find this.

Thanks! It's clear now.